packages/expo-video/ios/VideoModule.swift

```diff
@@ -73,12 +73,23 @@ public final class VideoModule: Module {
     }
 
     Class(VideoPlayer.self) {
-      Constructor { (source: String?) -> VideoPlayer in
-        if let source, let url = URL(string: source) {
-          let item = AVPlayerItem(url: url)
-          return VideoPlayer(AVPlayer(playerItem: item))
+      Constructor { (source: VideoSource) -> VideoPlayer in
+        let player = AVPlayer()
+        let videoPlayer = VideoPlayer(player)
```

Comment:
Have you thought about what we talked about some time ago, to make another class `VideoDRMPlayer` that inherits from `VideoPlayer` so that the latter doesn't have to know about DRM?

Reply:
I think I prefer the current version. 
I'm not sure what are the benefits of having a separate player for DRM sources, other than clean separation between types of players and slightly lower memory usage because we wouldn't be using the `contentKeyManager`.

The part that i don't like it that when calling `player.replace` and replacing a non-DRM source with a DRM source we'd have to replace the regular player with the `VideoDRMPlayer`. This would make the implementation more complicated and in my opinion negate the cleanliness" benefit of separating the players, so I'm not sure if it's worth it. Also `contentKeyManager` is lazy initialised so there will be no memory difference for `non-DRM` only players.